### ISSUE TITLE
qwen3-moe-forward-v1 ACTIVE_RUNTIME flip — operator-confirm cosine ≥ 0.99 vs HF FP16 reference (~60 GB download)

### ISSUE THREAD

**Issue by noahgift** [paiml/aprender#1584]
## Context

M32d **FUNCTIONALLY DISCHARGED** 2026-05-02 via #1228 squash `5235aaeb9` (Step 5 + 5b + 6 + 7 fix bundle: per-head Q/K RMSNorm + rope_theta default 10K → 1M for qwen3_moe + chat template no-think + traced sync). Output transitioned on lambda-vector RTX 4090 against the cached 17.3 GB Qwen3-Coder-30B-A3B-Instruct-Q4_K_M GGUF:
- pre-fix: `%%%%%%%%` (gibberish, repeated argmax)
- post-fix: `2 + 2 = 4` + multi-domain coherent answers (math/geography/translation/code)

## Remaining gate

The formal flip of `qwen3-moe-forward-v1` v1.4.0 **ACTIVE_ALGORITHM_LEVEL** → **ACTIVE_RUNTIME** is gated on the cosine ≥ 0.99 vs HF FP16 measurement at LM-head logits (**FALSIFY-QW3-MOE-PARITY-001**, contract v1.3.0).

## Operator-confirm dependency

`scripts/generate_qwen3_moe_fp16_logits.py` (#1129 squash `87a2a61c1`) downloads `Qwen/Qwen3-Coder-30B-A3B-Instruct` (~60 GB) and dumps `[batch, seq, vocab]` logits to JSON. Multi-device offload via `device_map="auto"`. ~30 min runtime on a 30B-A3B model.

## Test ready

- `crates/aprender-serve/tests/qwen3_moe_parity.rs::f_qw3_moe_parity_001` (#1130 squash `ce6ca4bb4`) — `#[ignore]`-gated; runs with `--include-ignored` once the fixture lands.
- Sibling `f_qw3_moe_argmax_parity_002` for llama.cpp argmax sanity (#1131 squash `9f93d02d9`, FALSIFY-QW3-MOE-PARITY-002, independent of HF fixture).

## Acceptance

- HF FP16 fixture present on disk
- `cargo test -p aprender-serve --test qwen3_moe_parity -- --include-ignored f_qw3_moe_parity_001` PASS
- Cosine ≥ 0.99 of APR `forward_qwen3_moe` LM-head logits vs HF FP16 reference
- Contract `qwen3-moe-forward-v1` v1.4.0 → v1.5.0 with status promoted to **ACTIVE_RUNTIME**

## Cross-refs

- Companion-repo: paiml/claude-code-parity-apr § Sub-extension 1 + § R9 risk
- Academic basis: arXiv:2210.17323 (GPTQ — quantization-aware reference comparison framework)

**Comment by noahgift** [comment]
## ✅ DISCHARGE EVIDENCE — F-QW3-MOE-PARITY-001 PASSED on lambda-vector RTX 4090 (2026-05-09)

Discovered via `find /mnt -name "*Qwen3-Coder-30B*"` that the HF FP16 weights have been on lambda-vector at `/mnt/nvme-raid0/models/Qwen3-Coder-30B-A3B-Instruct/` (57 GB across 16 safetensors shards, Mar 8 timestamps) for ~62 days — the "operator-confirm pending ~60 GB download" claim was stale.

### Live test results

```
cargo test --release -p aprender-serve --test qwen3_moe_parity -- \
  --include-ignored f_qw3_moe_parity_001_cosine_vs_hf_fp16

F-QW3-MOE-PARITY-001:
  elapsed       = 555.52868ms
  cos_sim       = 0.995384
  threshold     = 0.99
  apr_argmax    = 3555 (val = 22.3671)
  hf_argmax     = 3555 (" What")

test result: ok. 4 passed; 0 failed; 0 ignored
```

- **cos_sim 0.995384** ≥ 0.99 threshold by margin 0.0054
- **apr_argmax = hf_argmax = 3555 (" What")** — exact agreement on argmax
- APR forward elapsed: 555ms on 7-token prompt "What is 2+2?"

### Reproduction recipe

```bash
# 1. Generate HF FP16 fixture (~52s on lambda-vector RTX 4090)
cd /home/noah/src/aprender
uv run --with torch --with transformers --with accelerate \
  scripts/generate_qwen3_moe_fp16_logits.py \
  --model /mnt/nvme-raid0/models/Qwen3-Coder-30B-A3B-Instruct \
  --output crates/aprender-serve/tests/fixtures/qwen3_moe_fp16_logits_pos0.json

# 2. Build apr-cli release binary (~40s)
cargo build --release -p apr-cli

# 3. Run cosine test (~5s after fixture+build)
cargo test --release -p aprender-serve --test qwen3_moe_parity -- \
  --include-ignored f_qw3_moe_parity_001_cosine_vs_hf_fp16
```

Companion-side record: paiml/claude-code-parity-apr#95 (M109).

### Recommended follow-up here in aprender

1. Amend `aprender-contracts/contracts/qwen3-moe-forward-v1.yaml` v1.4.0 → v1.5.0 ACTIVE_ALGORITHM_LEVEL → **ACTIVE_RUNTIME** with `status_history` entry citing this discharge (companion repo PR #95 + cos_sim 0.995384).
2. Optionally: deprioritize FALSIFY-QW3-MOE-PARITY-002 (llama.cpp argmax) — transitive sibling no longer load-bearing because PARITY-001 directly proved `apr_argmax = hf_argmax`. Keep the test for regression coverage but don't gate ACTIVE_RUNTIME on it. (Note: `llama-cli` on Qwen3-Coder-30B-A3B-Instruct CPU-only hung at 99.9% single-CPU for 2 hrs without producing output even with `-ngl 999` GPU-offload flag — suspected MoE expert dispatch is CPU-bound in llama.cpp build #7746; that's an llama.cpp upstream issue, not an aprender issue.)
3. Closing this issue when the v1.5.0 ACTIVE_RUNTIME amendment merges.
